### PR TITLE
Make data model more flexible

### PR DESCRIPTION
**Pull request by mxxk**
So far, we have had a very well-defined and rigid data schema. Every meeting has a General Evaluator. Every meeting has up to two speakers. And so on... As the app moves forward, it is beneficial to relax some of these constraints and allow for flexible meeting roles. This PR is the first step in that direction.

Because current instances of the app are using the previous data model, this PR only makes new instances of the app use the new data model, and leaves the previous data model and corresponding React components which access it in place... for now. :wink:

After this PR is in place, the next steps are to:

1. Write a migration function to move data from the old model to the new.
2. Completely remove the old data model and React components.

Features
========

- Speech project selection and cancellation is **live!** When a document is opened in multiple places, these changes will get synced between them. :smile:
- New data model makes it possible to have custom roles in the future.
- React components are organized between multiple files.
- CSS modules isolate styles with auto-generates classnames.

### CODE REVIEW

**Comment by annyhe** [comment]
@mxxk I'm overloaded at work, @borncrusader please review :)

**Comment by mxxk** [comment]
> @mxxk I'm overloaded at work, @borncrusader please review :)

Understood @annyhe, thanks for letting me know. I forgot to mention that even a cursory glance 👀 at this PR to make sure there are no oversights would be greatly appreciated. 😄

Also you can rest assured, as the code is extensively tested for coexistence of old and new data (i.e. old Toastmasters Agenda apps continue to function, and new apps are created with the new data model). I also have ready the code used for migrating old to new data, so that's another data point to tell you this works (tested on a copy of the SFTM Signup Sheet):

image

File: src/components/roles.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const date = rootRecord.get('date');
+
+    return (
+        <table className={styles.roles}>
+            <tr>
+                <td>Meeting Date</td>
+                <td><PlainRichTextBox record={date} /></td>
+            </tr>
+            {
+                rootRecord
```
Comment (borncrusader) [inline src/components/roles.js]: Nice. :) I like this.

File: src/root.jsx [hunk 2]
```diff
@@ -56,12 +65,15 @@ class RootRecord extends quip.apps.RootRecord {
         timer: quip.apps.RichTextRecord,
         ahCounter: quip.apps.RichTextRecord,
         speakerSlot: quip.apps.RecordList.Type(SpeechSelect),
-    })
+    });
```
Comment (borncrusader) [inline src/root.jsx]: Props for good hygiene :)

**Pull request merged**